New script to build APKs for all channels (alpha / beta / release) and check version codes

maxme opened the pull request:
New script to build APKs for all channels (alpha / beta / release) and check version codes

tonyr59h commented:
The APK's are all being generated correctly, nice work!

I wish there was an easy way to check the version codes before building everything, but switching branches to grab them all first seems like overkill IMO.

Few things:
* Can we checkout the original branch after all the builds are complete? The original branch being whatever branch was checked-out when kicking off the script.
* Could we copy the generated APK's + build.log to a unique folder? Simply for convenience.
* `Wrong version codes (╯°□°）╯︵ ┻━┻` is a bit ambiguous (:+1: for the art though!). I think it's easy enough to find out what's wrong since the script is pretty straightforward, but we could provide detail in the message rather than forcing a new user to parse the script.

maxme commented:
> I wish there was an easy way to check the version codes before building everything, but switching branches to grab them all first seems like overkill IMO.

Yep but to be honest, this check shouldn't catch error often. When I do the release I'm very cautious of which version code I use for which version name / branch. So having it at the end of the script is probably good enough.

> Can we checkout the original branch after all the builds are complete? The original branch being whatever branch was checked-out when kicking off the script.

Done in b8c361c87e7f69c0e7e5b8192dff45e1e225e791

> Could we copy the generated APK's + build.log to a unique folder? Simply for convenience.

Moved to `build/`

> Wrong version codes (╯°□°）╯︵ ┻━┻ is a bit ambiguous (:+1: for the art though!). I think it's easy enough to find out what's wrong since the script is pretty straightforward, but we could provide detail in the message rather than forcing a new user to parse the script.

It was my intention, but I forgot the exit call - fixed in aff7cb6e78e2a081c937091cd292064df3a0e374

I've also added a check in 9d5cc3b695e3e398946e2d2fe00e09077e200cb2

tonyr59h commented:
:shipit:
